Sources/TensorFlow/Layers/Convolutional.swift

```diff
@@ -641,3 +641,84 @@ public extension DepthwiseConv2D {
             padding: padding)
     }
 }
+
+/// A layer for cropping tensors along the temporal dimension.
+public struct Cropping1D<Scalar: TensorFlowFloatingPoint>: Layer {
+    /// The cropping dimensions along the temporal axis.
+    @noDerivative public let cropping: (Int, Int)
+
+    /// Creates a cropping layer to trim the temporal dimension.
+    ///
+    /// - Parameter cropping: A tuple of two integers describing how many units should be trimmed
+    ///   off the beginning and end of the cropping dimension.
+    public init(cropping: (Int, Int) = (1, 1)) {
+        self.cropping = cropping
+    }
+
+    /// Creates a cropping layer to trim the temporal dimension.
+    ///
+    /// - Parameter cropping: An integer describing how many units should be symmetrically trimmed
+    ///   off the beginning and end of the cropping dimension. i.e. `(cropping, cropping)`.
+    public init(cropping: Int = 1) {
+        self.cropping = (cropping, cropping)
+    }
+
+    /// Returns the cropped input tensor according to the `cropping` dimensions specified
+    /// at initialization.
+    ///
+    /// - Parameter input: The 3D tensor to be cropped. Note the expected shape of the input is
+    ///   `[batch size, axis to crop, features]`.
+    /// - Returns: The cropped 3D tensor of shape `[batch, cropped axis, feature]`.
+    @differentiable
+    public func callAsFunction(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
+        let (begin, end) = cropping
+        return input.slice(
+            lowerBounds: [0, begin, 0],
+            upperBounds: [input.shape[0], input.shape[1] - end, input.shape[2]])
+    }
+}
+
+/// A layer for cropping tensors along spatial dimensions, e.g. image cropping.
+///
+/// `Cropping2D` can trim an input at the top, bottom, left, and right sides of
+/// its spatial dimensions.
+public struct Cropping2D<Scalar: TensorFlowFloatingPoint>: Layer {
+    /// The cropping dimensions along the height and width axes.
+    @noDerivative public let cropping: ((Int, Int), (Int, Int))
```

Comment:
We should compensate for weak type information. In this case, a property name that repeats the struct name and a nested tuple of integers do not meaningfully tell us anything about what the value represents, so the user always needs to go to the documentation.

Instead, I would suggest listing out the exact dimensions as _separate properties_ for clarity. Each property should have its own API documentation.

```suggestion
    @noDerivative public let top: Int
    @noDerivative public let bottom: Int
    @noDerivative public let left: Int
    @noDerivative public let right: Int
```